polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -3245,7 +3257,29 @@ where
 		if transition.is_local_approval() || newly_approved || !already_approved_by.unwrap_or(true)
 		{
 			// In all other cases, we need to write the candidate entry.
-			db.write_candidate_entry(candidate_entry);
+			db.write_candidate_entry(candidate_entry.clone());
```

Comment:
Looks like this clone could easily be avoided by changing the order?

Reply:
I second this we don't want this clone here since it is on the hotpath.